src/Templating/TemplateNameParser.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\BlockBundle\Templating;
+
+use Symfony\Bundle\FrameworkBundle\Templating\TemplateNameParser as FrameworkTemplateNameParser;
+
+/**
+ * @deprecated since 3.9, to be removed with 4.0.
```

Comment:
why? its new

Reply:
Same as above. But this class is `sonata.templating.name_parser` service